tests/phpunit/includes/testcase-domain.php

```diff
@@ -24,6 +27,100 @@ public static function wpTearDownAfterClass() {
 		$_SERVER = self::$server;
 	}
 
+	public function set_up() {
+		parent::set_up();
+
+		$this->filter_plugins_url();
+
+		$this->backup = array(
+			'REQUEST_URI' => $_SERVER['REQUEST_URI'],
+			'HTTP_HOST'   => $_SERVER['HTTP_HOST'],
+		);
+	}
+
+	public function tear_down() {
+		parent::tear_down();
+
+		$_SERVER['REQUEST_URI'] = $this->backup['REQUEST_URI'];
+		$_SERVER['HTTP_HOST']   = $this->backup['HTTP_HOST'];
+	}
+
+	protected function base_test_flags_urls( $curlang ) {
+		// Needed by {@see pll_requested_url()}.
+		$_SERVER['HTTP_HOST'] = wp_parse_url( $this->hosts[ $curlang->slug ], PHP_URL_HOST );
+
+		$frontend          = new PLL_Frontend( $this->links_model );
+		$frontend->curlang = $curlang;
+		$frontend->init();
+		$languages = $frontend->model->get_languages_list();
+
+		$this->assertCount( 3, $languages ); // @see `self::wpSetUpBeforeClass()`.
+
+		foreach ( $languages as $flag_language ) {
+			$code = 'en' === $flag_language->slug ? 'us' : $flag_language->slug;
+			$dir  = $this->is_directory ? '/sub' : '';
+			$this->assertSame(
+				$this->hosts[ $curlang->slug ] . "{$dir}/wp-content/plugins/polylang/flags/{$code}.png",
+				$flag_language->get_display_flag_url(),
+				"{$flag_language->name} flag URL with current language set to {$curlang->name} is wrong."
+			);
+		}
+	}
+
+	protected function set_directory() {
+		// Fake WP install in subdir.
+		update_option( 'siteurl', 'http://example.org/sub' );
+		update_option( 'home', 'http://example.org' );
+	}
+
+	/**
+	 * @ticket #1296
+	 * @see https://github.com/polylang/polylang/issues/1296.
+	 */
+	public function test_flags_urls_curlang_default() {
+		if ( $this->is_directory ) {
+			$this->set_directory();
+		}
+
+		$en = self::$model->get_language( 'en' );
+
+		$this->base_test_flags_urls( $en );
+	}
+
+	/**
+	 * @ticket #1296
+	 * @see https://github.com/polylang/polylang/issues/1296.
+	 */
+	public function test_flags_urls_curlang_secondary() {
```

Comment:
Since we have assertions based on the default language, maybe we should make sure to define it in `wpSetUpBeforeClass()` or `set_up()` to prevent porosity?
```php
self::$model->options['default_lang'] = 'en';
```

Reply:
Sure, done in ab4e60e1.